Filter/StringFilter.php

```diff
@@ -55,7 +55,7 @@ public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $data)
      *
      * @return bool
      */
-    private function getOperator($type)
+    protected function getOperator($type)
```

Comment:
Let's state the requirements cleary : the string filter should be case insensitive by default (because who wants case sensitive filtering ?), which means different operators should be injected into it, and implies some sort of guesser does this injection base on the current database platform (given by `Connection::getDatabasePlatform()`). Options to also have case sensitive filters might be added on another PR.